python/lib/client/dmod/client/request_clients.py

```diff
@@ -1,600 +1,823 @@
+            return response_type(success=False, reason=reason, message=msg)
+
+        await self._transport_client.async_send(data=str(request))
+        response_data = await self._transport_client.async_recv()
+        response_obj = response_type.factory_init_from_deserialized_json(json.loads(response_data))
+        if not isinstance(response_obj, response_type):
+            msg = f"{self.__class__.__name__} {originating_func_name} could not deserialize {response_type.__name__}" \
+                  f" from raw response data '{response_data}'"
+            raise DmodRuntimeError(msg)
         else:
-            # TODO: look at if there needs to be an addition to connection count, active connections, or something here
-            tmp = await self._async_acquire_new_session(
-                cached_session_file=self._cached_session_file
-            )
-            # logger.info("Session Info Return: {}".format(tmp))
-            return tmp
+            return response_obj
+
+    # TODO: better integrate uploading initial data into the create request itself
+    async def create_dataset(self, name: str, category: DataCategory, domain: DataDomain,
+                             upload_paths: Optional[List[Path]] = None, data_root: Optional[Path] = None,
+                             **kwargs) -> DatasetManagementResponse:
+        """
+        Create a dataset from the given parameters.
+
+        Parameters
+        ----------
+        name : str
+            The name for the dataset.
+        category : DataCategory
+            The category for the dataset.
+        domain : DataDomain
+            The defined domain for the dataset.
+        upload_paths : Union[Path, List[Path]]
+            List of paths of files to upload.
+        data_root : Optional[Path]
+            A relative data root directory, used to adjust the names for uploaded items.
+        kwargs
+
+        Returns
+        -------
+        DatasetManagementResponse
+            A response to the request for the service to create a new dataset, which may actually be a
+            ::class:`MaaSDatasetManagementMessage` depending on whether this instance uses auth.
+
+        Raises
+        -------
+        DmodRuntimeError
+            If the response from the service cannot be deserialized successfully to the expected response type.
 
-    def _process_data_download_iteration(self, raw_received_data: str) -> Tuple[bool, Union[DataTransmitMessage, MaaSDatasetManagementResponse]]:
+        See Also
+        -------
+        upload_to_dataset
+        uses_auth
         """
-        Helper function for processing a single iteration of the process of downloading data.
+        request = DatasetManagementMessage(action=ManagementAction.CREATE, domain=domain, dataset_name=name,
+                                           category=category)
+        create_response = await self._process_request(request=request, originating_func_name='create_dataset')
```

Comment:
Just a general comment. I get the purpose of `originating_func_name`, but it is raising my code smell alarms.

Reply:
Indeed, it was less that ideal. I've refactored things.

Reply:
Thanks!